cmd/minikube/cmd/start.go

```diff
@@ -141,6 +145,15 @@ func runStart(cmd *cobra.Command, args []string) {
 		glog.Errorln("Error starting the VM: ", err)
 		os.Exit(1)
 	}
+
+	// Set Proxy to as Shell Env
+	if viper.IsSet("http-proxy") || viper.IsSet("https-proxy") {
```

Comment:
>  Why I am actually not in favor to move registration bits and this one to StartHost because https://github.com/minishift/minishift/blob/master/cmd/minikube/cmd/start.go#L139 so whatever we move cluster.StartHost it will going to provision 3 time in case of any error. 

Ok, we might change the retry then. I don't see much use in this anyway. Unless we are covering for a error condition which occurs sporadically. 

Otherwise, we at least need to move the functionality of registration into a public method in cluster.